tests/protocol/test_wire.py

```diff
@@ -492,24 +430,20 @@ def test_get_ext_conf_without_uri(self, *args):
             self.assertIsNone(ext_conf.status_upload_blob_type)
             self.assertIsNone(ext_conf.artifacts_profile_blob)
 
-    def test_get_ext_conf_with_uri(self, *args):
-        mock_data = mockwiredata.WireProtocolData(mockwiredata.DATA_FILE)
-
-        with patch("azurelinuxagent.common.utils.restutil.http_get", side_effect=mock_data.mock_http_get):
-            with patch("azurelinuxagent.common.protocol.wire.CryptUtil", side_effect=mock_data.mock_crypt_util):
-                protocol = WireProtocol(WIRESERVER_URL)
-                # This call would populate the ext_conf with the test data and also mock the required certs
-                protocol.detect()
-                wire_protocol_client = protocol.client
-                ext_conf = wire_protocol_client.get_ext_conf()
-
-                self.assertEqual(1, len(ext_conf.ext_handlers.extHandlers))
-                self.assertEqual(2, len(ext_conf.vmagent_manifests.vmAgentManifests))
-                self.assertEqual("https://test.blob.core.windows.net/vhds/test-cs12.test-cs12.test-cs12.status?"
-                                 "sr=b&sp=rw&se=9999-01-01&sk=key1&sv=2014-02-14&sig=hfRh7gzUE7sUtYwke78IOlZOrTRCYvkec4hGZ9zZzXo",
-                                 ext_conf.status_upload_blob)
-                self.assertEqual("BlockBlob", ext_conf.status_upload_blob_type)
-                self.assertEqual(None, ext_conf.artifacts_profile_blob)
+    def test_get_ext_conf_with_uri(self):
+        with mock_wire_protocol(mockwiredata.DATA_FILE) as protocol:
+            protocol.detect()
+            wire_protocol_client = protocol.client
+            ext_conf = wire_protocol_client.get_ext_conf()
+
+            self.assertEqual(1, len(ext_conf.ext_handlers.extHandlers), "Unexpected number of extension handlers")
```

Comment:
what is the intention of this test? The name of the test doesn't tell me what it is trying to do, there are no comments explaining what is is trying to do, and the messages in the asserts don't help me there either.

Looking at the code, it gives me the impression that it is just a basic test for get_ext_conf() and all we are doing here is checking that the data returned by the function matches mockwiredata.DATA_FILE. If that is indeed the intention of the test a simple docstring might help the reader of this test. The "with_uri" part in the test name I cannot explain? Can you?

Reply:
Yes, the test is just asserting the values from the `ext_conf.xml` are properly populated. The name `with_uri` contrasts the name of the previous test, `without_uri`, where the extension config has no extensions set.

I will add a comment for both tests and improve the naming.

Reply:
Thanks. 

Let's also improve the assertion messages. Since that is indeed the intention of the test then "The number of handlers in the extension config does not match the test data: [{0}]" might be better than "Unexpected number of extension handlers" (where {0} is the list of handlers that the test fetched).

Similar for other asserts

Reply:
Initially I did that, but the objects in the list then show up as:
`AssertionError: 2 != 1 : Unexpected number of extension handlers: [[<azurelinuxagent.common.protocol.restapi.ExtHandler object at 0x7f71d4d20048>]]`

which I didn't think was valuable.

Reply:
Simple iteration over the handlers extracting the name would do it

Reply:
but if you think it's an overkill you can leave as is, but do review the other messages for improvements

Reply:
Updated.